src/api/services/item.service.ts

```diff
@@ -14,13 +14,44 @@ class ItemService {
   }
 
   async getAll() {
-    return this.fastify.db.select().from(itemTable).execute()
+    const items = await this.fastify.db
+      .select()
+      .from(itemTable)
+      .leftJoin(itemToCategoryTable, eq(itemTable.id, itemToCategoryTable.itemId))
+      .leftJoin(categoryTable, eq(itemToCategoryTable.categoryId, categoryTable.id))
+      .execute()
+
+    if (items.length === 0) return []
+
+    const result = items.reduce((acc, { item, category }) => {
+      if (!item || !category) return acc
+
+      const existingItem = acc.find((i) => i.id === item.id)
+
+      if (existingItem) existingItem.categoryId.push(category.id)
+      else acc.push({ ...item, categoryId: [category.id] })
```

Comment:
this is not correct. `categoryId` cannot be an array! change the name to `categories`!